src/components/v2/shared/FundingCycleConfigurationDrawers/TokenDrawer.tsx

```diff
@@ -0,0 +1,48 @@
+import { t } from '@lingui/macro'
+import FundingCycleDrawer from 'components/v2/shared/FundingCycleConfigurationDrawers/FundingCycleDrawer'
+import { useFundingCycleDrawer } from 'components/v2/shared/FundingCycleConfigurationDrawers/useFundingCycleDrawer'
+import UnsavedChangesModal from 'components/v2/shared/UnsavedChangesModal'
+
+import TokenForm from '../../V2Create/forms/TokenForm'
+
+export default function TokenDrawer({
+  visible,
+  onClose,
+  isCreate,
+}: {
+  visible: boolean
+  onClose: VoidFunction
+  isCreate?: boolean
+}) {
+  const {
+    handleDrawerCloseClick,
+    emitDrawerClose,
+    setFormUpdated,
+    unsavedChangesModalVisible,
+    closeModal,
+  } = useFundingCycleDrawer(onClose)
```

Comment:
fucking fresh ser. Such a good approach. We need to do more of this probably.

cc @johnnyd-eth @peripheralist

Reply:
Yeah nice love it